Quantum-Control-Applications/Superconducting/Single Flux Tunable Transmon/SSB mixer for readout/IQ_blobs.py

```diff
@@ -86,30 +86,20 @@
 else:
     qm = qmm.open_qm(config)
     job = qm.execute(singleshot)
-    res_handles = job.result_handles
-
-    Ie_handles = res_handles.get("Ie")
-    Qe_handles = res_handles.get("Qe")
-    Ie_handles.wait_for_values(1)
-    Qe_handles.wait_for_values(1)
-    Ig_handles = res_handles.get("Ig")
-    Qg_handles = res_handles.get("Qg")
-    Ig_handles.wait_for_values(1)
-    Qg_handles.wait_for_values(1)
+    # Get results from QUA program
+    results = fetching_tool(job, data_list=["Ie", "Qe", "Ig", "Qg"], mode="wait_for_all")
 
     # Live plotting
-    fig = plt.figure(figsize=(15, 15))
-    interrupt_on_close(fig, job)  #  Interrupts the job when closing the figure
-    while res_handles.is_processing():
-        I_e = Ie_handles.fetch_all()["value"]
-        Q_e = Qe_handles.fetch_all()["value"]
-        I_g = Ig_handles.fetch_all()["value"]
-        Q_g = Qg_handles.fetch_all()["value"]
-        plt.cla()
-        plt.scatter(I_g[: min(len(I_g), len(Q_g))], Q_g[: min(len(I_g), len(Q_g))], color="b", alpha=0.1)
-        plt.scatter(I_e[: min(len(I_e), len(Q_e))], Q_e[: min(len(I_e), len(Q_e))], color="r", alpha=0.1)
-        plt.axis("equal")
-        plt.xlabel("I")
-        plt.ylabel("Q")
-        plt.legend(["Ground", "Excited"])
-        plt.pause(0.1)
+    fig = plt.figure(figsize=(7, 5))
+
+    # Fetch results
+    Ie, Qe, Ig, Qg, iteration = results.fetch_all()
+    plt.cla()
+    plt.scatter(I_g[: min(len(I_g), len(Q_g))], Q_g[: min(len(I_g), len(Q_g))], color="b", alpha=0.1)
+    plt.scatter(I_e[: min(len(I_e), len(Q_e))], Q_e[: min(len(I_e), len(Q_e))], color="r", alpha=0.1)
+    plt.axis("equal")
+    plt.xlabel("I")
+    plt.ylabel("Q")
+    plt.legend(["Ground", "Excited"])
+
+    angle, threshold, fidelity, gg, ge, eg, ee = two_state_discriminator(I_g, Q_g, I_e, Q_e, b_print=True, b_plot=True)
```

Comment:
The `two_state_discriminator` does the plotting, so you would get here double the plots...
I also don't fully understand what you plot here

Reply:
Yes, but not in real-time...

Reply:
Ahh I see.
Then I would recommend adding a comment explaining it the different lengths.
Is live plotting really needed? This usually runs in 2 seconds...

Reply:
Ahh I see.
Then I would recommend adding a comment explaining it the different lengths.
Is live plotting really needed? This usually runs in 2 seconds...

Reply:
True I just did it because the guys in Finland wanted it it :)
I'll remove it now that we have the nice two_state_discriminator function